Gems/Archive/Code/Include/Archive/Tools/ArchiveWriterAPI.h

```diff
@@ -223,20 +228,20 @@ namespace Archive
         virtual ArchiveAddFileResult AddFileToArchive(AZ::IO::GenericStream& inputStream,
             const ArchiveWriterFileSettings& fileSettings = {}) = 0;
 
-        //! Used the span contents to add the file to the archive
+        //! The span contents is used to supply file data for the file to the archive
         virtual ArchiveAddFileResult AddFileToArchive(AZStd::span<const AZStd::byte> inputSpan,
             const ArchiveWriterFileSettings& fileSettings = {}) = 0;
 
         //! Searches for a relative path within the archive
         //! @param relativePath Relative path within archive to search for
         //! @return A token that identifies the Archive file if it exist
         //! if the with the specified path doesn't exist InvalidArchiveFileToken is returned
-        virtual ArchiveFileToken FindFile(AZ::IO::PathView relativePath) = 0;
+        virtual ArchiveFileToken FindFile(AZ::IO::PathView relativePath) const = 0;
```

Comment:
I was actually going to recommend const initially, but these days I'm on the fence about using const in early API iterations because of the situations when a dev may want to do something non-consty like say caching requests to FindFile or ContainsFile, or something else I'm not thinking of.  Thoughts?

Reply:
Well the API here for the IArchiveWriter is meant to allow access to the ArchiveWriter across DLL boundaries and be available as part of the `Archive.Editor.API` target.
It is not an abstract interface meant for other Gems or libraries to implement, so the API is made in mind with the current implementation. 

I prefer using `const` with these APIs as to the outside caller it should appear constant to them.
For a caching request, what should be done is to NOT make the function non-const, but instead add a `mutable` member that performs the caching

I am of the believe that the internal implementation details of an interface should not leak into the signature of the interface.